app/main.py

```diff
@@ -1,25 +1,71 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.max_amount = max_amount
+        self.min_amount = min_amount
+
+    def __set_name__(
+        self,
+        owner: type[object],
+        name: str
+    ) -> None:
+        self.public_name = name
```

Comment:
As for me, you don't need a public_name attribute here, so you can remove it but it isn't obligatory.